datafusion/functions-nested/src/map.rs

```diff
@@ -161,7 +174,19 @@ impl ScalarUDFImpl for MapFunc {
         &self.signature
     }
 
-    fn return_type(&self, arg_types: &[DataType]) -> datafusion_common::Result<DataType> {
+    fn return_type(
+        &self,
+        _arg_types: &[DataType],
+    ) -> datafusion_common::Result<DataType> {
+        internal_err!("map: return_type called instead of return_type_from_exprs")
+    }
+
+    fn return_type_from_exprs(
+        &self,
+        _args: &[Expr],
+        _schema: &dyn ExprSchema,
+        arg_types: &[DataType],
+    ) -> datafusion_common::Result<DataType> {
```

Comment:
Curiously, why are you using `return_type_from_exprs` instead? It seems that neither `args` nor `schema` will be used. I guess we can just use `return_type`.

Reply:
I chose to override this method because we were returing different return type based on input type. May want the refer the existing slt tests

Reply:
I reverted it back. I is misunderstood the requirement.